test/csuite/tsc_ratio/main.c

```diff
@@ -0,0 +1,64 @@
+#include <math.h>
+
+#include "test_util.h"
+
+/*
+ * main --
+ *     Initialize the library and if it determines a tick -> nsec ratio, test that ratio looks
+ *     reasonable.
+ */
+int
+main(int argc, char *argv[])
+{
+    static double NSEC_PER_SEC = 1.e9;
+
+    /* Maximum number of attempts to find the smallest error. */
+    static const int ATTEMPT_MAX = 3;
+
+    /* Maximum acceptable error as a fraction of measurement. */
+    static double ERR_TOLERANCE = 0.001;
```

Comment:
I've written tests like this, and had ongoing maintenance issues dealing with slow machines or slowness from competing VMs in a shared environment.  Things to consider: maybe be a little more tolerant?  Would things be particularly bad if we allowed 1% rather than 0.1 percent?  Maybe have a lot more attempts - say 20 attempts, spaced by one second sleeps in between?  And the test could exit on the first success, since we're really just looking for a minimum to pass?

Reply:
Yes I see I could see this being a potential problem. What do you think of an alternate test mechanism which takes  the existing timing loop and compares the result with register read? I think that approach should be more resistant to environment variation.

Reply:
Sure, that sounds promising.